src/main/java/org/candlepin/subscriptions/security/auth/AdminWithWhitelistedAccountOnly.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.security.auth;
+
+import org.candlepin.subscriptions.security.IdentityHeaderAuthenticationDetailsSource;
+
+import org.springframework.security.access.prepost.PreAuthorize;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A security annotation ensuring that the user has an admin role and has an account that has been
+ * white listed.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@PreAuthorize("hasRole('" + IdentityHeaderAuthenticationDetailsSource.ORG_ADMIN_ROLE + "') and " +
+    "@reportAccessService.providesAccessTo(authentication)")
+public @interface AdminWithWhitelistedAccountOnly {
```

Comment:
Rather than a new annotation, I'd rather just see this applied to `@AdminOnly`

Reply:
Ack. Will make the change. I wish spring security allowed putting multiple security annotations on a method. :/

Reply:
DONE